app/scripts/directives/metas/metas-actividades.js

```diff
@@ -38,247 +39,380 @@ angular.module('contractualClienteApp')
                 swal({
-                  title: '¡No hay actividades!',
-                  type: 'error',
-                  text: 'Las dependencias no están asociadas a la meta seleccionada.Por favor seleccione otra meta',
+                  title: "¡No hay Dependencias Seleccionadas!",
+                  type: "error",
+                  text: "Las dependencias no han sido seleccionadas",
                   showCloseButton: true,
-                  confirmButtonText: "CERRAR"
+                  confirmButtonText: "CERRAR",
                 });
-
-
               }
-            } else {
-              self.gridOptions.data = [];
-              self.gridApi.grid.modifyRows(self.gridOptions.data);
-              swal({
-                title: '¡No hay Dependencias Seleccionadas!',
-                type: 'error',
-                text: 'Las dependencias no han sido seleccionadas',
-                showCloseButton: true,
-                confirmButtonText: "CERRAR"
-              });
-            }
-
-          });
-        }
-
+            });*/
+        };
 
         // se observa cambios en actividades para seleccionar las respectivas filas en la tabla
-        $scope.$watch('actividades', function () {
-            $scope.actividades ? $scope.actividades.forEach(function (act) {
-              var tmp = self.gridOptions.data.filter(function (e) { return e.actividad_id !== act.ActividadId })
-              if (tmp.length > 0) {
-                self.gridApi.selection.selectRow(tmp[0]); //seleccionar las filas
-              }
-            }) : _;
-            self.actividades = $scope.actividades;
+        $scope.$watch("actividades", function () {
+          $scope.actividades
+            ? $scope.actividades.forEach(function (act) {
+                var tmp = self.gridOptions.data.filter(function (e) {
+                  return e.actividad_id !== act.ActividadId;
+                });
+                if (tmp.length > 0) {
+                  self.gridApi.selection.selectRow(tmp[0]); //seleccionar las filas
+                }
+              })
+            : _;
+          self.actividades = $scope.actividades;
         });
 
-        $scope.$watch('[d_metasActividades.gridOptions.paginationPageSize, d_metasActividades.gridOptions.data]', function () {
-          if ((self.gridOptions.data.length <= self.gridOptions.paginationPageSize || self.gridOptions.paginationPageSize === null) && self.gridOptions.data.length > 0) {
-            $scope.gridHeight = self.gridOptions.rowHeight * 2 + (self.gridOptions.data.length * self.gridOptions.rowHeight);
-            if (self.gridOptions.data.length <= 5) {
-              self.gridOptions.enablePaginationControls = false;
+        $scope.$watch(
+          "[d_metasActividades.gridOptions.paginationPageSize, d_metasActividades.gridOptions.data]",
+          function () {
+            if (
+              (self.gridOptions.data.length <=
+                self.gridOptions.paginationPageSize ||
```

Comment:
Puede que hagan falta los siguientes `()` para desambiguar?

```suggestion
              ( (self.gridOptions.data.length <=
                self.gridOptions.paginationPageSize) ||
```

(En cualquier caso, es posible que hagan falta un par de `()`, con el mero objetivo de clarificar)